artifacthub/library/pod-security-policy/forbidden-sysctls/1.0.0/template.yaml

```diff
@@ -51,3 +68,16 @@ spec:
         forbidden_sysctl(sysctl) {
             startswith(sysctl, trim(input.parameters.forbiddenSysctls[_], "*"))
         }
+
+        # * may be used to allow all sysctls
+        allowed_sysctl(sysctl) {
+            input.parameters.allowedSysctls[_] == "*"
+        }
+
+        allowed_sysctl(sysctl) {
+            input.parameters.allowedSysctls[_] == sysctl
+        }
+
+        allowed_sysctl(sysctl) {
+            startswith(sysctl, trim(input.parameters.allowedSysctls[_], "*"))
```

Comment:
I think this may unintentionally treat all `allowedSysctls` as having "*", we should filter to make sure that the allowedSysctl item includes `*` prior to attempting fuzzy matching. Also, since we are doing prefix matching, we should only remove the trailing `*`, not any leading `*`s.

I realize these bugs are present above for the disallowed list, but it's more of a security issue for "allowed sysctls".

IMO we should fix the disallowed list too, but not a blocker for me for this PR.

Reply:
Quite right. I fixed the rules, including one for `forbiddenSysctls`, in 140279af39b1376d8e028124b9a1be35e436d972